weatherbot/main.py

```diff
@@ -23,7 +23,7 @@ def main():
     puller = State("./puller_state")
     while True:
         updates = get_updates()
-        if updates['ok'] is True:
+        if updates['result'] == [] or updates["ok"] is False:
```

Comment:
It's very likely that response won't contain `result` key at all if error happened, so I'd leave only `updates["ok"] is False` check.